lib/streamlit/DeltaGenerator.py

```diff
@@ -1639,12 +1644,21 @@ def slider(
             # single variable
             current_value = current_value[0] if single_value else current_value
 
+        # Set format default.
+        if format is None:
+            if all_ints:
+                format = "%d"
+            else:
+                format = "%0.2f"
```

Comment:
Add a comment here saying something like:

> It would be great if we could guess the number of decimal places from the step`argument, but this would only be meaningful if step were a decimal. As a possible improvement we could make this function accept decimals and/or use some heuristics for floats.